client/FxAccountClient.js

```diff
@@ -905,6 +905,123 @@ define([
+   *   a URL that the client should be redirected to after handling the request
+   *   @param {String} [options.resume]
+   *   Opaque url-encoded string that will be included in the verification link
+   *   as a querystring parameter, useful for continuing an OAuth flow for
+   *   example.
+   *   @param {Object} [options.metricsContext={}] Metrics context metadata
+   *     @param {String} options.metricsContext.deviceId identifier for the current device
+   *     @param {String} options.metricsContext.flowId identifier for the current event flow
+   *     @param {Number} options.metricsContext.flowBeginTime flow.begin event time
+   *     @param {Number} options.metricsContext.utmCampaign marketing campaign identifier
+   *     @param {Number} options.metricsContext.utmContent content identifier
+   *     @param {Number} options.metricsContext.utmMedium acquisition medium
+   *     @param {Number} options.metricsContext.utmSource traffic source
+   *     @param {Number} options.metricsContext.utmTerm search terms
+   *   @param {String} [options.unblockCode]
+   *   Login unblock code.
+   * @return {Promise} A promise that will be fulfilled with JSON `xhr.responseText` of the request
+   */
+  FxAccountClient.prototype.sessionReauth = function (sessionToken, email, password, options) {
+    var self = this;
+    options = options || {};
+
+    return Promise.resolve()
+      .then(function () {
+        required(sessionToken, 'sessionToken');
+        required(email, 'email');
+        required(password, 'password');
+
+        return credentials.setup(email, password);
+      })
+      .then(
+        function (result) {
+          var endpoint = '/session/reauth';
+
+          if (options.keys) {
+            endpoint += '?keys=true';
+          }
+
+          var data = {
+            email: result.emailUTF8,
+            authPW: sjcl.codec.hex.fromBits(result.authPW)
+          };
+
+          if (options.metricsContext) {
+            data.metricsContext = metricsContext.marshall(options.metricsContext);
+          }
+
+          if (options.reason) {
+            data.reason = options.reason;
+          }
+
+          if (options.redirectTo) {
+            data.redirectTo = options.redirectTo;
+          }
+
+          if (options.resume) {
+            data.resume = options.resume;
+          }
+
+          if (options.service) {
+            data.service = options.service;
+          }
+
+          if (options.unblockCode) {
+            data.unblockCode = options.unblockCode;
+          }
+
+          if (options.originalLoginEmail) {
+            data.originalLoginEmail = options.originalLoginEmail;
+          }
+
+          if (options.verificationMethod) {
+            data.verificationMethod = options.verificationMethod;
+          }
+
+          return hawkCredentials(sessionToken, 'sessionToken',  HKDF_SIZE)
+            .then(function (creds) {
+              return self.request.send(endpoint, 'POST', creds, data);
```

Comment:
nit: @rfk if you want EVEN more testing you can take a look at https://github.com/mozilla/fxa-js-client/pull/217/files to make sure the correct options are sent

Reply:
Thanks!  I am *so* there for even more testing :-)